File: mvl/mvl/report/retirement_report/retirement_report.js
// Copyright (c) 2023, veeramayandi.p@groupteampro.com and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Retirement Report"] = {
	"filters": [
		{
			"fieldname":"from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			// "default":frappe.datetime.year_start(),
			"reqd": 1,
			
			"width": "100px"
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			// "default": frappe.datetime.year_end(),
			"reqd": 1,
			"width": "100px",
			on_change: function () {
				var from_date = frappe.query_report.get_filter_value('from_date')
				frappe.call({
					method: "mvl.mvl.report.monthwise_salary_register.monthwise_salary_register.get_to_date",
					args: {
						from_date: from_date
					},
					callback(r) {
						frappe.query_report.set_filter_value('to_date', r.message);
						frappe.query_report.refresh();
					}
				})
			}
		},
	]
};

File: mvl/mvl/report/retirement_report/retirement_report.py
# Copyright (c) 2023, Frappe Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

import frappe

def execute(filters=None):
	columns = get_columns(filters)
	data = get_data(filters)
	return columns, data

def get_columns(filters):
	columns = []
	columns += [
		("Employee") + ":Link/Employee:200",
		("Employee Name") + ":Data/:200",
		("Department") + ":Link/Department:200",
		("Designation") + ":Link/Designation:200",
		("Date Of Retirement") + ":Date/:100",
	]
	return columns

def get_data(filters):
	data = []
	sa = frappe.db.sql(""" select * from `tabEmployee` where status = "Active" and  date_of_retirement between '%s' and '%s'  """%(filters.from_date,filters.to_date),as_dict=True)
	# frappe.errprint(sa)
	for i in sa:
		frappe.errprint(i.employee)
		row = [i.name,i.employee_name,i.department,i.designation,i.date_of_retirement]
		data.append(row)
	return data
